### PR TITLE
PLU-328: [PAGINATE-3] refactor page titles

### PR DESCRIPTION
**Pull request by pregnantboy**
### Changes
- refactor existing pages to use shared compoenents PageTitle, DebouncedSearchInput and usePaginationAndFilter 
- move hamburger menu for small devices to replace plumber logo
- fix some css styling

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#758** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/758?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#773](https://github.com/opengovsg/plumber/pull/773) <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/773?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#774](https://github.com/opengovsg/plumber/pull/774) <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/774?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈
* **#757** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/757?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#756** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/756?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/src/exports/components.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export { default as ChooseConnectionSubstep } from '../components/ChooseConnecti
 export { default as ConditionalIconButton } from '../components/ConditionalIconButton'
 export { default as Container } from '../components/Container'
 export { default as ControlledAutocomplete } from '../components/ControlledAutocomplete'
+export { default as DebouncedSearchInput } from '../components/DebouncedSearchInput'
```
Comment (m0nggh) [inline packages/frontend/src/exports/components.ts:17]: actually do we need to add new ones to this list for it to work?

File: packages/frontend/src/components/ExecutionRow/index.tsx [hunk 2]
```diff
@@ -47,19 +47,19 @@ export default function ExecutionRow(props: ExecutionRowProps): ReactElement {
             templateAreas={{
               base: `
                 "apps arrow-container"
-                "title arrow-container"
+                "title title"
```
Comment (m0nggh) [inline packages/frontend/src/components/ExecutionRow/index.tsx:50]: wah this actually looks nicer thanks

File: packages/frontend/src/components/FlowRow/index.tsx [hunk 3]
```diff
@@ -103,7 +103,7 @@ export default function FlowRow(props: FlowRowProps): ReactElement {
               </VStack>
             </GridItem>
             <GridItem area="menu">
-              <Flex alignItems="center" gap={1.5}>
+              <Flex alignItems="center" gap={1.5} justifyContent="flex-end">
```
Comment (m0nggh) [inline packages/frontend/src/components/FlowRow/index.tsx:106]:
```suggestion
              <Flex alignItems="center" gap={1} justifyContent="flex-end">
```
felt like maybe a bit closer looks nicer but to ur discretion sir

File: packages/frontend/src/pages/Flows/index.tsx [hunk 4]
```diff
@@ -24,159 +20,113 @@ import EmptyFlows from './components/EmptyFlows'
...
-  const flows: IFlow[] = edges?.map(({ node }: { node: IFlow }) => node)
-  const hasFlows = flows?.length > 0
-  const isSearching = flowName !== '' || page !== 1
-
-  const isEmptyState = !hasFlows && !isSearching
-  const isEmptySearchResults = !loading && !hasFlows && isSearching
-  const hasPagination = pageInfo && pageInfo.totalCount > FLOW_PER_PAGE
+  const flows: IFlow[] = edges?.map(({ node }: { node: IFlow }) => node) ?? []
+  const hasPagination =
+    !loading && pageInfo && pageInfo.totalCount > FLOW_PER_PAGE
```
Comment (m0nggh) [inline packages/frontend/src/pages/Flows/index.tsx:88]:
```suggestion
    !loading && pageInfo?.totalCount > FLOW_PER_PAGE
```

File: packages/frontend/src/pages/Tiles/index.tsx [hunk 5]
```diff
@@ -32,42 +66,35 @@ export default function Tiles(): JSX.Element {
...
-  const hasNoTiles =
-    !isTileListLoading && tilesToDisplay?.length === 0 && input.trim() === ''
-  const hasNoSearchResults =
-    !isTileListLoading && tilesToDisplay?.length === 0 && input.trim() !== ''
+  const { pageInfo, edges } = data?.getTables ?? {}
+  const tilesToDisplay = edges?.map(({ node }) => node) ?? []
+
+  const hasNoUserTiles = tilesToDisplay.length === 0 && !isSearching
+  const hasPagination =
+    !loading && pageInfo && pageInfo.totalCount > TILES_PER_PAGE
```
Comment (m0nggh) [inline packages/frontend/src/pages/Tiles/index.tsx:74]:
```suggestion
    !loading && pageInfo?.totalCount > TILES_PER_PAGE
```

File: packages/frontend/src/pages/Flows/index.tsx [hunk 6]
```diff
@@ -24,159 +20,113 @@ import EmptyFlows from './components/EmptyFlows'
...
-        )}
-
-        {hasPagination && (
-          <Flex justifyContent="center" mt={6}>
-            <Pagination
-              currentPage={pageInfo?.currentPage}
-              onPageChange={handlePageChange}
-              pageSize={FLOW_PER_PAGE}
-              totalCount={pageInfo?.totalCount}
+    <Container py={9} overflowY={'hidden'}>
```
Comment (m0nggh) [inline packages/frontend/src/pages/Flows/index.tsx]: image

can i check why is the `overflowY` added ah, cus now with one pipe only the context menu is hidden leh

File: packages/frontend/src/exports/components.ts (hunk 1 above)
Comment (pregnantboy) [reply packages/frontend/src/exports/components.ts:17]: not really. but in case admin panel need

**Review by m0nggh — APPROVED** [review]
thks for the refactor, lgtm sir

**Comment by pregnantboy** [comment]
### Merge activity

* **Oct 14, 11:45 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/758).
* **Oct 14, 11:45 AM EDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/758).

**Pull request merged**